relay-server/src/actors/processor.rs

```diff
@@ -1013,42 +1014,60 @@ impl EnvelopeProcessorService {
         }
 
         envelope.retain_items(|item| match item.ty() {
-            ItemType::Profile => match relay_profiling::expand_profile(&item.payload()) {
-                Ok(payload) => {
-                    if payload.len() <= self.config.max_profile_size() {
-                        item.set_payload(ContentType::Json, payload);
-                        true
-                    } else {
+            ItemType::Profile => {
+                let contexts = match state.event.value_mut() {
+                    Some(event) if event.ty.value() == Some(&EventType::Transaction) => {
+                        event.contexts.value_mut().as_mut()
```

Comment:
Don't we also want to create `contexts` here if it does not exist? I.e. https://github.com/getsentry/relay/blob/57aacf07c7665e8d995e79648d98e8fadbabec8d/relay-general/src/protocol/replay.rs#L283

Reply:
Not too sure what the right move here is, but when I had this, a lot of integration tests started failing. I'm not sure why, maybe you can help shed some lights on this?

Also, I'm not sure how realistic it is for a transaction to have 0 context, so I thought this was acceptable.

Reply:
Transactions will always have contexts in practice (the trace context is even required). Still, `get_or_insert_with` is more explicit, and also improves the quality of our tests where we had broken fixtures.

The CI failures occurred because of [this change](https://github.com/getsentry/relay/pull/1801/files/a3da2f056fc136d876b0cfe3c4101a0aa2db5140#diff-ff64e5542d3e0a29fab7ed5550d22511a6ca00cefaae3bc9a8de7b158499d792R980). `process_profiles` runs on _every_ envelope, even the ones without a profile. You unconditionally create an empty error event (without an event ID), which does not pass assertions during storage. This creates an error which caused test failures. 

You should check for the existence of an event, but within the event, you can always create the context.